aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -47,6 +49,15 @@ module aptos_framework::account {
     struct RotationCapability has drop, store { account: address }
     struct SignerCapability has drop, store { account: address }
 
+    struct OriginatingAddress has key, store {
+        address_map: Table<address, address>,
+    }
+
+    struct RotationProof has drop {
+        originator: address, // originating address
```

Comment:
I'm thinking we should probably add the current sequence number (account.sequence_number) as the first element.